src/helpers/radspec.js

```diff
@@ -44,57 +62,64 @@ export default (eth, evaluator) =>
     const methodId = data.substr(0, 10)
     const fn = functions[methodId]
 
-    // If function is not a known function, execute fallback checking Parity's on-chain signature registry
+    // If function is not a known function
     if (!fn) {
-      // Even if we pass the ETH object, if it is not on mainnet it will use Aragon's ETH mainnet node
-      // As the registry is the only available on mainnet
-      const registry = new MethodRegistry({ networkId: '1', eth })
-      const result = await registry.lookup(methodId)
-
-      if (result) {
-        const { name } = registry.parse(result)
+      try {
+        // Try checking on-chain signature registry
+        const registry = new MethodRegistry({
+          registryAddress,
+          provider,
+          network: registryAddress
+            ? undefined
+            : (await provider.getNetwork()).chainId
+        })
+        const result = await registry.lookup(methodId)
+        const { name } = parse(result)
         return {
           type: 'string',
           value: name // TODO: should we decode and print the arguments as well?
         }
-      } else {
+      } catch {
+        // Try fetching 4bytes API
+        const { results } = await ethers.utils.fetchJson(
```

Comment:
If we can't resolve the signature with the on-chain registry we try fetching the 4bytes API. Otherwise, we finally default to an unknown function.